docs/lint-tools-cpp.md

```diff
@@ -0,0 +1,73 @@
+# Linting for C++ projects
+
+We use `clang-format` for formatting and `clang-tidy` for code analysis.
+
+## General usage
+
+C++ projects should use the [.clang-format] and [.clang-tidy] config files as follows (assuming
+`yscope-dev-utils` was added to the project at `tools/yscope-dev-utils`):
+
+1. Add a setup step to symlink the files to the project root.
+   * You can use [symlink-cpp-lint-configs.sh].
+2. Create project-specific config files to override any settings as necessary (see below).
+3. Run `clang-format` and `clang-tidy` using the config files.
+
+## Creating project-specific config files
+
+* Create a `.clang-format` or `.clang-tidy` file in the project subdirectory (e.g. `src`) where the
+  settings are necessary.
+* In each config file, set the relevant option (see the tool-specific sections below) to inherit
+  from the config file in the parent directory.
+
+### clang-format
+
+In the project-specific config file:
+
+* Set `BasedOnStyle: InheritParentConfig` to inherit from config file in the parent directory.
```

Comment:
Does this only work if the project specific .clang-format is in a subdirectory of the root?

Reply:
It can be in any parent (grandparent, etc.) directory.